### PR TITLE
Bundle improvements

### PR DESCRIPTION
**Pull request by vinicius73**
Even if the rollup was installed it was not used in the package build process.
This PR solves this and adds other improvements.

Related to: #295
Fix: #5 

- Update dependencies
- Remove babel
  - Typescript already is able to generate code that is in compliance with es5
- Remove `package.json` from generated code.
  - Use [`@rollup/plugin-replace`](https://www.npmjs.com/package/@rollup/plugin-replace) to handle this.
- Add [`ESM`](https://nodejs.org/docs/latest-v14.x/api/esm.html) support
  - Improve native usage support
  - Improve bundle size
- Fix invalid typescript typings export ([dist/lib](https://unpkg.com/browse/@nextcloud/event-bus@2.1.1/dist/lib/) -> [dist](https://github.com/nextcloud/nextcloud-event-bus/blob/master/package.json#L6))
- Remove unnecessary files from [npm package](https://unpkg.com/browse/@nextcloud/event-bus@2.1.1/)

### CODE REVIEW

File: lib/ProxyBus.ts
Comment (ChristophWurst) [inline lib/ProxyBus.ts:23]: I've learned something new

**Review by ChristophWurst — APPROVED** [review]
Looks good to me

Let's have a second review on this big change

**Review by juliushaertl — APPROVED** [review]
Good from my side as well 👍

File: tsconfig.json
Comment (vinicius73) [reply tsconfig.json:6]: This is handled by rullup ts plugin @artonge 

```ts
typescript({
  tsconfig: './tsconfig.json',
  compilerOptions: { target: 'es5' },
})
```

image

This is the default bundle target.

File: lib/ProxyBus.ts
Comment (vinicius73) [reply lib/ProxyBus.ts:23]: @artonge it is not a real global variable.
It will be replaced in bundle file.

image

File: tsconfig.json
Comment (st3iny) [reply tsconfig.json:6]: It will probably not go through babel again because we exclude node_modules by default. I think we should keep targetting ES6 here.

Comment (st3iny) [reply tsconfig.json:6]: Would it be possible to put the target into `tsconfig.json` directly? This would certainly reduce confusion.

File: .github/workflows/node-test.yml
Comment (skjnldsv) [inline .github/workflows/node-test.yml:9]: Please keep the Node/test Ci output like other repositories :)

```yml
name: Node

on:
  push:
    branches:
      - '*'

jobs:
  test:
```

Comment (skjnldsv) [inline .github/workflows/node-test.yml:9]: Please keep the Node/test Ci output like other repositories :)
Also, keep pull_requests

```yml
name: Node

on:
  pull_request:
  push:
    - master

jobs:
  test:
```

File: tsconfig.json
Comment (vinicius73) [reply tsconfig.json:6]: Now this package generates 2 output files.
- `dist/index.js` -> previous commonjs target, also es5 compliance.
- `dist/index.esm.js` -> new esm target, without es5 transpiling.

And it is addressed in `package.json`

```json
"main": "dist/index.js",
"module": "dist/index.esm.js",
"types": "dist/index.d.ts",
"exports": {
  "import": "./dist/index.esm.js",
  "require": "./dist/index.js"
},
```

Webpack, rollup, vite, parcel, skypack and any other bundle/builder can now read these options and choose the better target option.

Even an older bundler will keep everything working because we steel using commonjs as default.

File: .github/workflows/node-test.yml
Comment (skjnldsv) [inline .github/workflows/node-test.yml:14]: Since we run a matrix, could you add a summary check so it's easier to maintain a branch protection?

https://github.com/nextcloud/.github/blob/3a0592a45b5d58b558eca89f414c0933c09e0044/workflow-templates/lint-php.yml#L38-L49

Comment (vinicius73) [reply .github/workflows/node-test.yml:14]: Done 👍🏼 
After this merge, looks good add eslint and prettier to run keep the code style consistent

File: lib/ProxyBus.ts
Comment (vinicius73) [reply lib/ProxyBus.ts:23]: It will run in the browser environment @skjnldsv, not in node like webpack config script.

I am only able to change it from rollup config, but there are some issues to use `process.env` and ESM syntax (like we are using in `rollup.config.js`)

Unfortunately, ESM and CJS are very incompatible and generate a lot of problems when we mix those things...

Comment (skjnldsv) [reply lib/ProxyBus.ts:23]: > It will run in the browser environment @skjnldsv, not in node like webpack config script.

Yes, if you define as a global variable in rollup, it's good :)
https://www.npmjs.com/package/@rollup/plugin-replace

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**
